cleaner kubernetes.yaml generation

**Description**
while working and comparing our yaml with example yaml i noticed a few things that could make ours a bit nicer:

- [ ] don't add quotes unless needed (i.e. [MINIMIZE_QUOTES](https://www.javadoc.io/doc/com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/2.7.5/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.Feature.html#MINIMIZE_QUOTES) )
- [ ] don't generate labels/annotations for vcs and similar if those values are not available, i..e. "<<unknown>>" looks  noisy

i.e. generate this:

```yaml
---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    app.quarkus.io/build-timestamp: "2020-05-13 - 20:16:21 +0000"
  labels:
    app.kubernetes.io/name: jugchdemo
    app.kubernetes.io/version: 1.0-SNAPSHOT
  name: jugchdemo
```

instead of 

```yaml
---
apiVersion: "v1"
kind: "ServiceAccount"
metadata:
  annotations:
    app.quarkus.io/vcs-url: "<<unknown>>"
    app.quarkus.io/build-timestamp: "2020-05-13 - 20:16:21 +0000"
    app.quarkus.io/commit-id: "<<unknown>>"
  labels:
    app.kubernetes.io/name: "jugchdemo"
    app.kubernetes.io/version: "1.0-SNAPSHOT"
  name: "jugchdemo"
```

@iocanel when trying to use dekorate `0.12.1` I am getting:

```
[ERROR] Failed to execute goal on project quarkus-kubernetes-deployment: Could not resolve dependencies for project io.quarkus:quarkus-kubernetes-deployment:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.dekorate:kubernetes-annotations:jar:noapt:0.12.1, io.dekorate:openshift-annotations:jar:noapt:0.12.1, io.dekorate:knative-annotations:jar:noapt:0.12.1: Failure to find io.dekorate:kubernetes-annotations:jar:noapt:0.12.1 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -144,7 +144,7 @@
         <aws-alexa-sdk.version>2.30.0</aws-alexa-sdk.version>
         <azure-functions-java-library.version>1.3.0</azure-functions-java-library.version>
         <kotlin.version>1.3.72</kotlin.version>
-        <dekorate.version>0.11.9</dekorate.version>
+        <dekorate.version>0.12.1</dekorate.version>
         <maven-artifact-transfer.version>0.10.0</maven-artifact-transfer.version>
         <jline.version>2.14.6</jline.version>
         <maven-invoker.version>3.0.1</maven-invoker.version>

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
@@ -370,16 +370,21 @@ class KubernetesProcessor {
         }
 
         ScmInfo scm = project.getScmInfo();
-        String vcsUrl = scm != null ? scm.getUrl() : Annotations.UNKNOWN;
-        String commitId = scm != null ? scm.getCommit() : Annotations.UNKNOWN;
+        String vcsUrl = scm != null ? scm.getUrl() : null;
+        String commitId = scm != null ? scm.getCommit() : null;
 
         //Dekorate uses its own annotations. Let's replace them with the quarkus ones.
         session.resources().decorate(target, new RemoveAnnotationDecorator(Annotations.VCS_URL));
         session.resources().decorate(target, new RemoveAnnotationDecorator(Annotations.COMMIT_ID));
         //Add quarkus vcs annotations
-        session.resources().decorate(target,
-                new AddAnnotationDecorator(new Annotation(QUARKUS_ANNOTATIONS_COMMIT_ID, commitId)));
-        session.resources().decorate(target, new AddAnnotationDecorator(new Annotation(QUARKUS_ANNOTATIONS_VCS_URL, vcsUrl)));
+        if (commitId != null) {
+            session.resources().decorate(target,
+                    new AddAnnotationDecorator(new Annotation(QUARKUS_ANNOTATIONS_COMMIT_ID, commitId)));
+        }
+        if (vcsUrl != null) {
+            session.resources().decorate(target,
+                    new AddAnnotationDecorator(new Annotation(QUARKUS_ANNOTATIONS_VCS_URL, vcsUrl)));
+        }
 
         if (config.isAddBuildTimestamp()) {
             session.resources().decorate(target, new AddAnnotationDecorator(new Annotation(QUARKUS_ANNOTATIONS_BUILD_TIMESTAMP,
@@ -737,8 +742,8 @@ class KubernetesProcessor {
         Project project = FileProjectFactory.create(artifactPath.toFile());
         BuildInfo buildInfo = new BuildInfo(app.getName(), app.getVersion(),
                 "jar", project.getBuildInfo().getBuildTool(),
+                artifactPath.toAbsolutePath().toString(),
                 artifactPath,
-                project.getBuildInfo().getClassOutputDir(),
                 project.getBuildInfo().getResourceDir());
 
         return new Project(project.getRoot(), buildInfo, project.getScmInfo());

--- a/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeContainerImageTest.java
+++ b/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeContainerImageTest.java
@@ -54,7 +54,7 @@ public class KnativeContainerImageTest {
                 Files.readAllLines(kubernetesDir.resolve("knative.yml"), StandardCharsets.UTF_8).toArray(new String[0]));
 
         for (String yamlFile : yamlFiles) {
-            if (yamlFile.contains("\"Service\"")) {
+            if (yamlFile.contains("Service") && !yamlFile.contains("ServiceAccount")) {
                 return DeserializationUtil.MAPPER.readValue(yamlFile, Service.class);
             }
         }

